addons/fleet/models/fleet_vehicle.py

```diff
@@ -0,0 +1,549 @@
+    image_small = fields.Binary(related='model_id.make_id.image_small', string="Logo (small)", store=True)
+    contract_renewal_due_soon = fields.Boolean(compute='_compute_get_contract_reminder',
+                                               search='_search_contract_renewal_due_soon',
+                                               string='Has Contracts to renew')
+    contract_renewal_overdue = fields.Boolean(compute='_compute_get_contract_reminder',
+                                              search='_search_get_overdue_contract_reminder',
+                                              string='Has Contracts Overdue')
+    contract_renewal_name = fields.Text(compute='_compute_get_contract_reminder', string='Name of contract to renew soon')
+    contract_renewal_total = fields.Integer(compute='_compute_get_contract_reminder',
+                                            string='Total of contracts due or overdue minus one')
+    car_value = fields.Float(help='Value of the bought vehicle')
+
+    _sql_constraints = [('unique_chassis_number', 'unique(vin_sn)', 'Same Chassis Number is already exists.')]
+
+    @api.one
+    @api.depends('license_plate')
+    def _compute_vehicle_name(self):
+        self.name = ('%s' + ' / ' + '%s' + ' / ' + '%s') % (self.model_id.make_id.name, self.model_id.name, self.license_plate)
+
+    @api.multi
+    def _compute_get_contract_reminder(self):
+        LogContract = self.env['fleet.vehicle.log.contract']
+        current_date = fields.Datetime.from_string(fields.Date.context_today(self))
+        for vehicle in self:
+            overdue = False
+            due_soon = False
+            total = 0
+            name = ''
+            for contract in LogContract.search([('vehicle_id', '=', vehicle.id), ('state', 'in', ('open', 'toclose')), ('expiration_date', '!=', None)],
+                                                limit=1, order='expiration_date asc'):
+                due_time = fields.Datetime.from_string(contract.expiration_date)
+                diff_time = (due_time-current_date).days
+                if diff_time < 0:
+                    overdue = True
+                    total += 1
+                if diff_time < 15 and diff_time >= 0:
+                    due_soon = True
+                    total += 1
+                if overdue or due_soon:
+                    name = contract.cost_subtype_id.name
+
+            vehicle.contract_renewal_overdue = overdue
+            vehicle.contract_renewal_due_soon = due_soon
+            vehicle.contract_renewal_total = (total - 1)  # we remove 1 from the real total for display purposes
+            vehicle.contract_renewal_name = name
+
+    @api.multi
+    def _compute_count_odometer(self):
+        Odometer = self.env['fleet.vehicle.odometer']
+        odometer_count = Odometer.read_group([('vehicle_id', '=', self.id)], fields=['vehicle_id'], groupby=['vehicle_id'])
```

Comment:
It is not safe to use self.id if you decorate your method with @api.multi, would you replace your code with,

odometer_count = self.env['fleet.vehicle.odometer'].read_group([('vehicle_id', 'in', self.ids)], fields=['vehicle_id'], groupby=['vehicle_id'])
vehicle_dict = dict((record['vehicle_id'][0], record['vehicle_id_count']) for record in odometer_count)
for vehicle in self:
    vehicle.odometer_count = vehicle_dict.get(vehicle.id)

Do similar stuff for all your *_count fields wherever search_count is used.